docs/upgrade/UPGRADE-unreleased.md

```diff
@@ -101,6 +101,43 @@ There you can find links to upgrade notes for other versions too.
     - modify the functional test `\Tests\ShopBundle\Functional\Component\Redis\RedisFacadeTest` so it creates `RedisFacade` using the two arrays and add a new test case `testNotCleaningPersistentClient`
         - you can copy-paste the [`RedisFacadeTest`](https://github.com/shopsys/project-base/blob/master/tests/ShopBundle/Functional/Component/Redis/RedisFacadeTest.php) from `shopsys/project-base`
 - implement `createFromIdAndName(int $id, string $name): FriendlyUrlData` method in your implementations of `FriendlyUrlDataFactoryInterface` as the method will be added to the interface in `v8.0.0` version ([#948](https://github.com/shopsys/shopsys/pull/948))
+- use aliases of index and build version in index name for better usage when deploying
+    - use method `ElasticsearchStructureManager::getCurrentIndexName` or `ElasticsearchStructureManager::getAliasName` instead of `ElasticsearchStructureManager::getIndexName` when calling a query to elasticSearch in order to always target the right index
+        - fix test `FilterQueryTest::createFilter`
+        ```diff
+        -   $elasticSearchIndexName = $elasticSearchStructureManager->getIndexName(1, self::ELASTICSEARCH_INDEX);
+        +   $elasticSearchIndexName = $elasticSearchStructureManager->getAliasName(1, self::ELASTICSEARCH_INDEX);
+        ```
+    - run `php phing product-search-recreate-structure` to generate new indexes with aliases
+    - delete your old indexes in elasticsearch
+    - fix test `ProductSearchExportRepositoryTest::getExpectedStructureForRepository`
+        ```diff
+        -   shortDescription
+        +   short_description
+        ```
+    - use method `ElasticsearchStructureManager::deleteCurrentIndex` instead of `ElasticsearchStructureManager::deleteIndex` as it was deprecated
+    - if you have extended `ElasticsearchStructureManager` in `services.yml` you'll need call `setBuildVersion` and to send build version to it
+        ```diff
+          Shopsys\FrameworkBundle\Component\Elasticsearch\ElasticsearchStructureManager:
+              arguments:
+                  - '%shopsys.elasticsearch.structure_dir%'
+                  - '%env(ELASTIC_SEARCH_INDEX_PREFIX)%'
+        +     calls:
+        +         - method: setBuildVersion
+        +           arguments:
+        +               - '%build-version%'
+        ```
+    - use method `ElasticsearchStructureManager::getCurrentIndexName` or `ElasticsearchStructureManager::getAliasName` instead of `ProductElasticsearchRepository::getIndexName` as it was deprecated
+    - change every number in mapping in `*.json` to string in order to be able to check definition for changes
```

Comment:
So even the `settings.index.number_of_shards` and `settings.index.number_of_replicas`? Why haven't you changed them yourself?

Maybe a bit better approach would be to recursively traverse the definition yourself during the need-to-update check and cast all integers you encounter to strings. That would be automatic, not as brittle and wouldn't require a change in the structure.